components/user-service/src/main/java/edu/jhuapl/dorset/ldapuserservice/LdapUserService.java

```diff
@@ -0,0 +1,250 @@
+        env.put(Context.SECURITY_PRINCIPAL, this.ldapBindDn);
+        env.put(Context.SECURITY_CREDENTIALS, this.ldapPassword);
+
+        try {
+            this.ctx = new InitialDirContext(env);
+        } catch (NamingException ex) {
+            throw new NamingException("Context could not be created. " + ex.getMessage());
+        }    
+        
+        this.setContext(this.ctx);
+    }
+
+    /**
+     * 
+     * LdapUserService
+     * 
+     * @param context Context object that is instantiated with LDAP server information
+     * @param userAttributes String array containing User attributes that is used to generate LDAP
+     *        search controls
+     */
+    public LdapUserService(DirContext context, String[] userAttributes) {
+        this.ctx = context;
+
+        this.ldapServer = null;
+        this.ldapSearchBase = null;
+        this.ldapPassword = null;
+        this.ldapBindDn = null;
+        this.ldapFilterAttribute = null;
+
+        this.userAttributes = userAttributes;
+
+        this.users = new HashMap<String, User>();
+    }
+
+    @Override
+    public String create(User user) throws UserException {
+        throw new UnsupportedOperationException("LdapUserService does not implement Create.");
+    }
+
+    @Override
+    public void delete(String userName) {
+        throw new UnsupportedOperationException("LdapUserService does not implement Delete.");
+    }
+
+    @Override
+    public User getUser(String userName) {
+        User user = null;
+
+        // check if user exists in the hash map
+        user = this.users.get(userName);
+
+        if (user == null) {
+
+            Properties props = new Properties();
+            props.setProperty("userName", userName);
+
+            try {
+                this.retrieve(props);
+            } catch (UserException e) {
+                logger.debug("User not found: " + userName);
+                return null;
+            }
+            user = this.users.get(userName);
+        }
+
+        return user;
+    }
+    
+    @Override
+    public String retrieve(Properties properties) throws UserException {
+        User user = new User();
+        String userName = properties.getProperty("userName");
+
+        String key;
+        Attribute value;
+        
+        SearchControls searchControls = getSearchControls(this.userAttributes);
+        ctx = getContext();
+
+        try {
+            NamingEnumeration<SearchResult> answer = ctx.search(this.ldapSearchBase,
+                            this.ldapFilterAttribute + "=" + userName, searchControls);
+            if (answer.hasMore()) {
+                Attributes attrs = answer.next().getAttributes();
+                NamingEnumeration<String> attrsKeys = attrs.getIDs();
+                while (attrsKeys.hasMoreElements()) {
+                    key = attrsKeys.next();
+                    value = attrs.get(key);
+                    user.setUserInformation(key, value.toString());
```

Comment:
Should we provide a way to map from LDAP keys to the keys used in the User object?